src/cdk/layout/media-matcher.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Injectable} from '@angular/core';
+import {Platform} from '@angular/cdk/platform';
+
+/**
+ * Global registry for all dynamically-created, injected style tags.
+ */
+let styleElementForWebkitCompatibility: {[key: string]: HTMLStyleElement} = {};
+
+@Injectable()
+export class MediaMatcher {
+  /** The internal matchMedia method to return back a MediaQueryList like object. */
+  private _matchMedia: (query: string) => MediaQueryList;
+
+  constructor(private platform: Platform) {
+    this._matchMedia = this.platform.isBrowser ?
+      (<any>window).matchMedia.bind((<any>window)) :
```

Comment:
Why does this need `bind`?

Reply:
We are unable to bind the `window.matchMedia` method as a property of our custom object as it results in an `illegal invocation`.
So we instead `bind` it to window so that its execution scope is always as expected

Reply:
Add a comment mentioning this?